src/index.ts

```diff
@@ -20,14 +28,26 @@ const getPostCssConfigPlugins = (directory: string) => {
   }
 };
 
-function init({ typescript: ts }: { typescript: typeof tsModule }) {
+const init: server.PluginModuleFactory = ({ typescript: ts }) => {
   let _isCSS: isCSSFn;
 
-  function create(info: ts.server.PluginCreateInfo) {
+  function create(info: server.PluginCreateInfo): LanguageService {
     const logger = createLogger(info);
     const directory = info.project.getCurrentDirectory();
     const compilerOptions = info.project.getCompilerOptions();
 
+    const languageServiceHost = {} as Partial<LanguageServiceHost>;
+
+    const languageServiceHostProxy = new Proxy(info.languageServiceHost, {
+      get(target, key: keyof LanguageServiceHost) {
+        return languageServiceHost[key]
+          ? languageServiceHost[key]
+          : target[key];
+      },
+    });
+
+    const languageService = createLanguageService(languageServiceHostProxy);
```

Comment:
Should this be using the ts parameter? If tsserver is run via VS Code, it may not match the version in node_modules.

Reply:
(Generally I'd suggest `type` importing only for plugins to avoid this mismatch)

Reply:
Also, you probably only want to import from tsserverlibrary, as it's a different bundle entirely.